### PR TITLE
github: Add workflows for code formatting and unix port build/test.

### PR DESCRIPTION
**Pull request by dpgeorge**
Travis CI has recently limited the amount of build time available for open-source projects. It's limited to 10k credits/month, which is about 1000 minutes, or around 65 builds for this repository.  That's not enough builds/month and is the reason that CI has stopped today (near the end of the month).

For more background see:
- https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
- https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

So I'd propose to switch all of the Travis jobs to GitHub Actions.  This PR starts with code formatting and the unix coverage build.

### CODE REVIEW

**Comment by stinos** [comment]
I can't find the issue at the moment, but it was about suggestions like putting the meat of build instructions in .yml files in shell scripts instead, allowing a) to just run them (instead of copy-pasting from the .yml which I've personally done way more than once) b) makes transitioons like this simple because you don't have to copy hundreds of lines since the CI builds just call the shell scripts. Wouldn't it be worth doing that right now?

**Comment by dpgeorge** [comment]
> putting the meat of build instructions in .yml files in shell scripts instead

Yes, that does make sense.  And maybe use a Makefile (or even Python script??) rather than a shell script so it can have all the targets in there.  We already require make and Python anyway for all builds.

I was aiming to get something done quickly because it really does block everything else...

**Comment by stinos** [comment]
> I was aiming to get something done quickly because it really does block everything else...

I get that makes it a bit peculiar to now select what exactly would work best (make/python/...). On the other hand even if it would run under make/python, plain shell scripts are still the lowest common denomitaor so I'd personally still use it anyway (also because if needed the shell script can install prerequisites like make/python, not the other way around:)

**Comment by stinos** [comment]
Overall I really like where this is going!
Not to nag or push you but are the `apt` and related commands also going to end up in ci.sh or similar?

**Comment by dpgeorge** [comment]
> Overall I really like where this is going!

Thanks!   It's a bit painful, but slowly getting there.

> Not to nag or push you but are the `apt` and related commands also going to end up in ci.sh or similar?

I wasn't planning on it, mainly because it's not always Ubuntu that these commands will be run on.  It would be complex to support multiple OS's.  Maybe an improvement for the future?

**Comment by dpgeorge** [comment]
This is working quite well now.  Still a few more jobs to add (eg esp32, bare-arm, unix nanbox) but otherwise it's getting close, and any further feedback on the approach used here is appreciated.

**Comment by stinos** [comment]
> I wasn't planning on it, mainly because it's not always Ubuntu

Fair enough, but on the other hand it's mainly CI and those are usually Ubuntu.

**Comment by dpgeorge** [comment]
All CI tasks from Travis have now been converted to GitHub Actions, so this PR should be good to go in, and to replace Travis completely.

> are the `apt` and related commands also going to end up in ci.sh or similar?

This is now done.  It did clean it up quite a lot!

**Comment by stinos** [comment]
Nice!

File: .github/workflows/ports_cc3200.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - 'drivers/**'
+      - 'ports/cc3200/**'
+
+jobs:
+  cc3200_build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install packages
+      run: sudo apt-get install gcc-arm-none-eabi libnewlib-arm-none-eabi
```
Comment (stinos) [inline .github/workflows/ports_cc3200.yml:21]: This is the only one which doesn't have it's setup step in ci.sh?

Comment (dpgeorge) [reply .github/workflows/ports_cc3200.yml]: Right, good catch, now fixed.  I also factored this generic gcc-arm-none-eabi set up into a reusable function.

**Pull request merged**